ledger/ledger-on-posix-filesystem/src/main/scala/com/daml/ledger/on/filesystem/posix/FileSystemLedgerReaderWriter.scala

```diff
@@ -193,14 +183,27 @@ class FileSystemLedgerReaderWriter private (
 object FileSystemLedgerReaderWriter {
   type Index = Int
 
-  private val StartOffset: Index = 0
+  private val StartIndex: Index = 0
 
-  def apply(
+  def owner(
       ledgerId: LedgerId,
       participantId: ParticipantId,
       root: Path,
-  )(implicit executionContext: ExecutionContext): Future[FileSystemLedgerReaderWriter] = {
-    val ledger = new FileSystemLedgerReaderWriter(ledgerId, participantId, root)
-    ledger.createDirectories().map(_ => ledger)
-  }
+  )(implicit executionContext: ExecutionContext): ResourceOwner[FileSystemLedgerReaderWriter] =
+    for {
+      dispatcher <- ResourceOwner.forCloseable(
+        () =>
+          Dispatcher(
+            "posix-filesystem-participant-state",
+            zeroIndex = StartIndex,
+            headAtInitialization = StartIndex,
+        ))
+      participant <- ResourceOwner.successful(
+        new FileSystemLedgerReaderWriter(ledgerId, participantId, root, dispatcher))
+      _ = participant.createDirectories()
+    } yield {
+      val participant = new FileSystemLedgerReaderWriter(ledgerId, participantId, root, dispatcher)
```

Comment:
Why is this duplicated in the for-comprehension and the yield block?

Reply:
Refactoring fail. Will fix.